pkg/proxy/healthcheck/healthcheck_test.go

```diff
@@ -131,8 +132,10 @@ type healthzPayload struct {
 func TestServer(t *testing.T) {
 	listener := newFakeListener()
 	httpFactory := newFakeHTTPServerFactory()
+	fakeClock := testingclock.NewFakeClock(time.Now())
+	hs := newProxierHealthServer(listener, httpFactory, fakeClock, "127.0.0.1:10256", 10*time.Second, nil, nil)
```

Comment:
The `interface` in my other comment means you don't have to do this fake-clock stuff (which I find hard to reason about and is not germane to this test)

```diff
diff --git a/pkg/proxy/healthcheck/healthcheck_test.go b/pkg/proxy/healthcheck/healthcheck_test.go
index dd3e9b76976..8db3feaecfb 100644
--- a/pkg/proxy/healthcheck/healthcheck_test.go
+++ b/pkg/proxy/healthcheck/healthcheck_test.go
@@ -129,13 +129,20 @@ type healthzPayload struct {
 	CurrentTime string
 }
 
+type fakeHealthChecker struct {
+	value bool
+}
+
+func (fake fakeHealthChecker) IsHealthy() bool {
+	return fake.value
+}
+
 func TestServer(t *testing.T) {
 	listener := newFakeListener()
 	httpFactory := newFakeHTTPServerFactory()
-	fakeClock := testingclock.NewFakeClock(time.Now())
-	hs := newProxierHealthServer(listener, httpFactory, fakeClock, "127.0.0.1:10256", 10*time.Second, nil, nil)
+	proxyChecker := &fakeHealthChecker{true}
 
-	hcsi := newServiceHealthServer("hostname", nil, listener, httpFactory, []string{}, hs)
+	hcsi := newServiceHealthServer("hostname", nil, listener, httpFactory, []string{}, proxyChecker)
 	hcs := hcsi.(*server)
 	if len(hcs.services) != 0 {
 		t.Errorf("expected 0 services, got %d", len(hcs.services))
@@ -354,13 +361,13 @@ func TestServer(t *testing.T) {
 	testHandler(hcs, nsn4, http.StatusOK, 6, t)
 
 	// fake a temporary unhealthy proxy
-	hs.oldestPendingQueued.Store(hs.clock.Now().Add(-hs.healthTimeout - time.Second))
+	proxyChecker.value = false
 	testHandlerWithHealth(hcs, nsn2, http.StatusServiceUnavailable, 0, false, t)
 	testHandlerWithHealth(hcs, nsn3, http.StatusServiceUnavailable, 7, false, t)
 	testHandlerWithHealth(hcs, nsn4, http.StatusServiceUnavailable, 6, false, t)
 
 	// fake a healthy proxy
-	hs.oldestPendingQueued.Store(hs.clock.Now())
+	proxyChecker.value = true
 	testHandlerWithHealth(hcs, nsn2, http.StatusServiceUnavailable, 0, true, t)
 	testHandlerWithHealth(hcs, nsn3, http.StatusOK, 7, true, t)
 	testHandlerWithHealth(hcs, nsn4, http.StatusOK, 6, true, t)
```